### DISCUSSION TITLE
Does `async-graphql` work with Relay?

### DISCUSSION THREAD

**Discussion by satvikpendem** [async-graphql/async-graphql#1268] [Q&A] [answered]
Or would I have to use another client library like Apollo?

**Comment by aidangilmore** [comment, marked as answer]
Yes, but you need to handle the node interface yourself (or you can use a crate that does that for you)

**Reply by satvikpendem** [reply]
Thanks, do you have an example or could you tell me more about the node interface, not sure what that means exactly? And which crate would handle that for me?

**Reply by aidangilmore** [reply]
The Node interface comes from the [GraphQL Global Object Identification Specification](https://relay.dev/graphql/objectidentification.htm), which is required by Relay to be implemented. There is a library for supporting this without too much manual work, [async-graphql-relay ](https://lib.rs/crates/async-graphql-relay). The library is currently unmaintained but still works.

**Reply by satvikpendem** [reply]
Thanks, I'll look into that crate. If you have any examples of it being used with async-graphql, I'd love to know, but no worries if not.

**Reply by aidangilmore** [reply]
Here are the examples from async-graphql-relay

https://github.com/oscartbeaumont/async-graphql-relay/tree/d51715d6069aac4c81eeb7581d5f416290e631f5/example
